modflow_devtools/netcdf_schema.py

```diff
@@ -0,0 +1,206 @@
+        return v
+
+    @field_validator("modflow_model", mode="before")
+    @classmethod
+    def validate_modflow_model(cls, v: Any, info: ValidationInfo) -> Any:
+        tokens = v.split(":")
+        if len(tokens) != 2:
+            raise ValueError(f"Invalid modflow_model attribute: {v}")
+        modeltype = tokens[0].strip()
+        if modeltype[-1].isdigit():
+            modeltype = modeltype[:-1]
+        info.context["modeltype"] = modeltype  # type: ignore
+        info.context["modelname"] = tokens[1].strip()  # type: ignore
+        return v
+
+
+class NetCDFPackageParam(BaseModel):
+    param: str = Field()
+    shape: list[str] = Field(default_factory=list)
+    attrs: "NetCDFParamAttrs"
+    encodings: "NetCDFParamEncodings"
+    varname: str = Field()
+    numeric_type: str = Field()
+
+    @field_validator("param", mode="before")
+    @classmethod
+    def validate_param(cls, v: Any) -> Any:
+        """
+        validate package parameter
+        """
+        tokens = v.split("/")
+        if len(tokens) != 3:
+            raise ValueError(f"Invalid param format: {v}")
+        model = tokens[0]
+        package = tokens[1]
+        param = tokens[2]
+        dfn = get_dfn(f"{model}-{package}")
+
+        blocks = ["griddata", "period"]
+        if not any(blk in dfn for blk in blocks):
+            raise ValueError(
+                f"griddata/period blocks not found in package type {package}"
+            )
+        if not any(param in dfn[blk] if blk in dfn else False for blk in blocks):
+            if (
+                "stress_period_data" not in dfn["period"]
+                or param not in dfn["period"]["stress_period_data"]["item"]["fields"]
+            ):
+                raise ValueError(f"Param {param} not found in package {package}")
+
+        for b in blocks:
+            if b in dfn and param in dfn[b]:
+                if not dfn[b][param]["netcdf"]:
+                    raise ValueError(f"Not a netcdf param: '{param}'")
+        return v
+
+    @field_validator("attrs", mode="before")
+    @classmethod
+    def validate_attrs(cls, v: Any, info: ValidationInfo) -> Any:
+        """
+        validate parameter attributes dictionary
+        """
+        v = {k.lower(): v.lower() if isinstance(v, str) else v for k, v in v.items()}
+        param = info.data.get("param")
+        shape = info.data.get("shape")
+        mesh = info.context.get("mesh")  # type: ignore
+        if mesh is not None and "z" in shape and "layer" not in v:  # type: ignore
+            raise AssertionError(f"Expected layer attribute for mesh param '{param}'")
+        if (
+            param is not None
+            and param.split("/")[2] == "aux"
+            and "modflow_iaux" not in v
+        ):
+            raise AssertionError(
+                f"Expected modflow_iaux attribute for aux param '{param}'"
+            )
+        return v
+
+
+class NetCDFParamAttrs(BaseModel):
+    modflow_input: str = Field()
```

Comment:
is there a rule for when an attribute not defined by NetCDF is prefixed with `modflow_...`? just wondering why `modflow_input` and `modflow_iaux` are but `layer` isn't

Reply:
Probably need to talk about this more- clearly internal, modflow specific details have the `modflow_` prefix, while attributes that are already recognizable in other contexts (`layer`) are used as such.  At least that's the idea.